### PR TITLE
Enable use_proxy feature

### PR DESCRIPTION
**Pull request by methadata**
Now, the module can be used inside a network that requires an HTTP proxy to access the Internet.
Just define use_proxy to true and configure http_proxy and https_proxy vars accordingly in the inventory file or in the playbook that makes use of this module.

e.g. in the inventory.yml file would be:
```yaml
xtesting:
  hosts:
    127.0.0.1:
      ansible_user: xtesting
      ansible_python_interpreter: auto
  vars:
    http_proxy: http://192.168.1.100:8080
    https_proxy: http://192.168.1.100:8080
    use_proxy: true
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+```yaml
+---
+xtesting:
+  hosts:
+    127.0.0.1:
+      ansible_user: xtesting
+      ansible_python_interpreter: auto
+  vars:
+    http_proxy: http://10.113.55.36:8080
```
Comment (collivier) [inline README.md:53]: usually 127.0.0.1 in doc

File: defaults/main.yml [hunk 2]
```diff
@@ -8,6 +8,10 @@ project: xtesting
 db_project: "{{ project }}"
 use_slave:  # true if slaves are deployed
 user_mail: root@localhost
+use_proxy: false
+
+# control if install software package and services
+deploy_base_infra: true
```
Comment (collivier) [inline defaults/main.yml:14]: At firt glance, I would prefer false by default.
As for privileges, I would prefer the safer way by default (the user does the operations by hand)

File: tasks/cachet.yml [hunk 3]
```diff
@@ -9,6 +9,11 @@
   delay: 10
   when:
     - cachet_deploy
+  environment:
```
Comment (collivier) [inline tasks/cachet.yml:12]: I would consider it as useless here

File: tasks/docker.yml [hunk 4]
```diff
@@ -1,9 +1,73 @@
 ---
-- name: Installing docker
+- name: Installing Python docker module requirements
```
Comment (collivier) [inline tasks/docker.yml:2]: Yes Installing docker is confusing. Well catch.
Installing docker SDK? or docker SDK for Python

File: tasks/docker.yml [hunk 5]
```diff
@@ -1,9 +1,73 @@
...
+    https_proxy: "{{ https_proxy if use_proxy else '' }}"
+
+- name: populate service ansible_facts
+  service_facts:
+
+- name: Ensure group "docker" exists
+  become: yes
+  group:
+    name: docker
+    state: present
```
Comment (collivier) [inline tasks/docker.yml:22]: A couple of when deploy_base_infra are missing, right?

File: tasks/jenkins.yml [hunk 6]
```diff
@@ -9,12 +9,17 @@
...
+    http_proxy: "{{ http_proxy if use_proxy else '' }}"
+    https_proxy: "{{ https_proxy if use_proxy else '' }}"
+
 - name: Starting Jenkins
   become: true
   docker_container:
     name: jenkins
     image: ollivier/xtesting-jenkins
-    pull: '{{ docker_pull }}'
+    # pull: '{{ docker_pull }}'
```
Comment (collivier) [inline tasks/jenkins.yml:22]: former local modification, right?

File: tasks/jenkins.yml [hunk 7]
```diff
@@ -26,31 +31,37 @@
...
     - jenkins_create_jobs
+
 - name: Loading Jenkins jobs
   command: |
     jenkins-jobs --conf \
       {{ tmp_dir }}/jenkins_jobs.ini update {{ tmp_dir }}/{{ project }}.yaml
   register: exit_code
   changed_when: exit_code.rc == 0
+  environment:
+    no_proxy: "{{ ipaddress }}"
```
Comment (collivier) [inline tasks/jenkins.yml:64]: to be removed

File: tasks/requirements.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
+---
+- name: Pre-taks | populate service ansible_facts
+  service_facts:
+
+- name: Pre-taks | Install pip (Debian)
+  become: yes
+  apt:
+    name:
+    - python-pip
+    - python3-pip
```
Comment (collivier) [inline tasks/requirements.yml:10]: we could rather autodetect the python version in used.

File: tasks/main.yml [hunk 9]
```diff
@@ -1,4 +1,6 @@
 ---
+- include_tasks: requirements.yml
+  when: deploy_base_infra
```
Comment (collivier) [inline tasks/main.yml:3]: or per task. Sounds good

File: tasks/testapi.yml [hunk 10]
```diff
@@ -16,39 +17,48 @@
   register: docker_response
   when:
     - testapi_deploy
+
+
```
Comment (collivier) [inline tasks/testapi.yml:21]: blank line

File: tasks/testapi.yml [hunk 11]
```diff
@@ -16,39 +17,48 @@
...
   when:
     - docker_response.changed
+
 - name: Checking if pod is already registered
   uri:
     url: '{{ testapi_url }}/pods/{{ project }}'
     status_code:
       - 200
       - 404
+    use_proxy: no
```
Comment (collivier) [inline tasks/testapi.yml:34]: I would consider it must be removed. We could consider that you may uuse proxy to reach an external TestDB (OPNFV for instance)

**Review by collivier — COMMENTED** [review]
Thx a lot. Would you mind removing the blank lines?
It seems you prefer blank lines more than me :)

I szz a couple of minor enhancements but it sounds in a good shape.

File: README.md (hunk 1 above)
Comment (methadata) [reply README.md:53]: I would suggest to use an inventory, even in command line/ This way, the ansible module would point always to "all" (or even better: xtesting group), and then specify which hosts to include in that group by using an inventory file or directly in CLI.

File: tasks/cachet.yml (hunk 3 above)
Comment (methadata) [reply tasks/cachet.yml:12]: PIP makes use of the proxy, as it needs it to access to Internet to get the pip packages.
So, I think is mandatory to have this here.

File: tasks/docker.yml (hunk 4 above)
Comment (methadata) [reply tasks/docker.yml:2]: This is the Python library for the Docker Engine API (installing via pip). This pip module is needed for Ansible to use the docker.container module, so I think it makes sense to install it before using docker with Ansible.

File: tasks/docker.yml (hunk 5 above)
Comment (methadata) [reply tasks/docker.yml:22]: not needed, as it's already specified in the "include" statement in `tasks/main.yml` ;-)

File: tasks/jenkins.yml (hunk 6 above)
Comment (methadata) [reply tasks/jenkins.yml:22]: yes. fixed

File: tasks/requirements.yml (hunk 8 above)
Comment (methadata) [reply tasks/requirements.yml:10]: well, yes but for new releases of the module :)

File: tasks/testapi.yml (hunk 11 above)
Comment (methadata) [reply tasks/testapi.yml:34]: ok

File: tasks/docker.yml (hunk 4 above)
Comment (collivier) [reply tasks/docker.yml]: My comment was only about the task name

**Comment by collivier** [comment]
I will slightly update it in a next change. It's already a good first step.
Thank you for your contribution!

**Pull request merged**